core/primitives/src/views.rs

```diff
@@ -457,6 +457,33 @@ pub struct NetworkGraphView {
     pub next_hops: HashMap<PeerId, Vec<PeerId>>,
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct LabeledEdgeView {
+    pub peer0: u32,
```

Comment:
are the edge directed or undirected? If they are directed, please rename peer0 and peer1 to peer_from and peer_to or some form that shows where the direction of the edge points to.

Reply:
Edges are directed. The way I see it EdgeView isn't meant to do anything smart, but rather just dumps the fields in Edge as-is for display on the frontend.

Reply:
if it's undirected then both edge1 and edge2 below should be equal
edge1.peer0 = a, edge1.peer1 = b should be equal to edge2.peer0=b, edge2.peer1=a. 
Just a mathematical nit, someone unfamiliar with the code looking at this code would have trouble figuring out if it's undirected or directed. Maybe useful to leave a comment at the least or rename the edges to be LabeledUndirectedEdgeView.

Reply:
There is an invariant hidden somewhere that in an edge peer0 < peer1 and it is enforced in runtime - that is on the real edge. IMO it's fine to have the UI structs to be a simple dump of the real structures used in networking. As always, +1 to adding comments though :)

Reply:
@soonnear the pattern here is that `SomeObjectView` is built from `SomeObject`, with the View containing the fields which we wish to pass to the debug UI. The priority for View is on accurate visibility into the system for debugging purposes. Constraints of the type you are describing make sense at the Object level, not at the View level.

Furthermore in this case peer0 < peer1 is already enforced at the Edge object level. If the Object has an invalid state, we would actually want to see that rather than masking it via some non-trivial View constructor.